lib/ffmpeg/transcoderbase.py

```diff
@@ -21,8 +21,8 @@ def before(self):
     self.refctx = []
     self.post_validate = lambda: None
     self.hwdevice = f"hw:{get_media().render_device}"
-    self.csc = "I420"
-    self.tonemap = None
+    self.format = "NV12"
+    self.hdr = False
```

Comment:
The vpp class and transcoder test classes are completely separate and treat csc differently.  The csc in vpp tests implies we use the hw-based filter/plugin that is directly responsible for it and avoid any implicit conversions.

The "csc" option in transcode was/is only useful/effective in "tonemap" cases.  Now, the option is moved to the "output" list, renamed to "format" and still only used in tonemap cases.  It is still a csc operation, but does not imply any specific plugin to do it. 
 Therefore, we will not see any functional change in our current test sets.

Later, I will refactor it to allow it to be used in any output case, not just tonemap.

Reply:
The vpp class and transcoder test classes are completely separate and treat csc differently.  The csc in vpp tests implies we use the hw-based filter/plugin that is directly responsible for it and avoid any implicit conversions.

The "csc" option in transcode was/is only useful/effective in "tonemap" cases.  Before, the first command-line (i.e. the actual xcode operation we are validating) always used NV12 as the tonemap output.  Then, csc format was only applied when we ran the second, intermediate, command-line to dump the xcode result for the metric check.  Now, with this PR, we use the output config "format" (aka csc) directly at the tonemap filter, as it should be.

Since csc was only effective for tonemap cases, we should not see any functional change in our other cases.

Later, I will refactor output "format" to be used in any output case, not just tonemap.  But, we don't have any cases configured right now that need it.

Reply:
In other words, this refactor fixes the strange tonemap case that previously did:
1. decode HDR P010 -> TONEMAP NV12 -> encode SDR NV12
2. decode SDR NV12 -> CSC P010 -> YUV P010

Now, it does:
1. decode HDR P010 -> TONEMAP P010 -> encode SDR P010
2. decode SDR P010 -> YUV P010